packages/cactus-test-plugin-htlc-eth-besu/src/test/typescript/integration/plugin-htlc-eth-besu/new-contract-endpoint.test.ts

```diff
@@ -0,0 +1,191 @@
+  t.comment(`HttpServer1 AddressInfo: ${JSON.stringify(addressInfo)}`);
+  const node1Host = `http://${addressInfo.address}:${addressInfo.port}`;
+  t.comment(`Cactus Node 1 Host: ${node1Host}`);
+
+  const pluginRegistry = new PluginRegistry({});
+  pluginRegistry.add(keychain);
+  pluginRegistry.add(connector);
+  const configService = new ConfigService();
+  const apiServerOptions = configService.newExampleConfig();
+  const pluginOptions: IPluginHtlcEthBesuOptions = {
+    logLevel,
+    instanceId: uuidv4(),
+    pluginRegistry,
+  };
+
+  const factoryHTLC = new PluginFactoryHtlcEthBesu({
+    pluginImportType: PluginImportType.LOCAL,
+  });
+
+  const pluginHtlc = await factoryHTLC.create(pluginOptions);
+  pluginRegistry.add(pluginHtlc);
+  apiServerOptions.configFile = "";
+  apiServerOptions.apiCorsDomainCsv = "*";
+  apiServerOptions.apiPort = addressInfo.port;
+  apiServerOptions.cockpitPort = 0;
+  apiServerOptions.apiTlsEnabled = false;
+  const config = configService.newExampleConfigConvict(apiServerOptions);
+  const apiServer = new ApiServer({
+    httpServerApi: httpServer,
+    config: config.getProperties(),
+    pluginRegistry,
+  });
+
+  await apiServer.start();
+  t.comment("Start server");
+  test.onFinish(() => apiServer.shutdown());
+  const configuration = new Configuration({
+    basePath: node1Host,
+  });
+  t.comment("Setting configuration");
+
+  const api = new DefaultApi(configuration);
+  t.comment("Api generated");
+
+  //Deploy contract
+  const deployOut = await pluginHtlc.initialize(
+    keychainId,
+    accountRef,
+    privateKeyRef,
+    dataTest.estimated_gas,
+    Web3SigningCredentialType.PRIVATEKEYHEX,
+    connectorId,
+  );
+
+  t.comment(deployOut.transactionReceipt.contractAddress!);
+  t.ok(deployOut, "deployContract is truthy OK");
+
+  const hashTimeLockAddress = deployOut.transactionReceipt
+    .contractAddress as string;
+  const balance = await web3.eth.getBalance(firstHighNetWorthAccount);
+  t.comment(balance);
+
+  // Test for 200 valid response test case
+  const bodyObj: NewContractObj = {
+    contractAddress: hashTimeLockAddress,
+    inputAmount: 10,
+    outputAmount: 0x04,
+    expiration: dataTest.expiration,
+    hashLock: dataTest.hashLock,
+    receiver: dataTest.receiver,
+    outputNetwork: "BTC",
+    outputAddress: "1AcVYm7M3kkJQH28FXAvyBFQzFRL6xPKu8",
+    keychainId: keychainId,
+    connectorId: connectorId,
+    accountRef: accountRef,
+    privateKeyRef: privateKeyRef,
+    credentialType: Web3SigningCredentialType.PRIVATEKEYHEX,
+    gas: dataTest.estimated_gas,
+  };
+  const res = await api.newContract(bodyObj);
+  t.comment("Getting result");
+  t.equal(res.status, 200);
+
+  const balance2 = await web3.eth.getBalance(firstHighNetWorthAccount);
+  t.comment(balance2);
+  t.equal(
+    parseInt(balance),
+    parseInt(balance2) - 10,
+    "Balance of account is OK",
+  );
+
+  t.ok("End test 1");
+
+  // Test for 500 not found test case
```

Comment:
@AzaharaC Code 500 should be returned for unknown errors. If the server validated the request and found that it had problems (such as something not being found that the request referred to as something in existence or something being in an invalid format) then a 400 or some other 4xx code could better indicate to end users that they need to fix something in their request.
In general 5xx codes mean that the backend is "at fault" meaning that the developer who sent the request has no action need, but in this case they do (which can be expressed in general with some 4xx codes that in general means that the backend thinks that the request sender is at fault)

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

Reply:
Done